rfc/0016-block-instantiation-syntax/0016-block-instantiation-syntax.md

```diff
@@ -0,0 +1,248 @@
+    - ` : <block>` models an anonymous instance of `<block>`, reusing the same syntax creates a new instance
+    - `<block>` models a singleton instance of `<block>`, reusing the same syntax reuses the same instance
+
+#### 3. Allow only pipes and block instances in pipelines
+The following explanations use blocks as an example but map directly to **Constraints** and **Valuetypes**. That is, constraints and valuetypes can also only be defined outside of pipelines and are instantiated inside of pipelines.
+
+- Pipelines may only contain pipes and block instances, everything else has to be described outside of a pipeline
+
+New syntax (minimal cars example):
+```
+// Pipeline containing block instantiations
+pipeline CarsPipeline {
+	myExtractor : CarsCSVExtractor 
+    -> myInterpreter : CarsTableInterpreter
+		-> myLoader : CarsLoader;
+}
+
+// Block definitions outside of pipeline
+block CarsCSVExtractor oftype CSVExtractor {
+  url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+  enclosing: '"';
+}
+block CarsTableInterpreter oftype TableInterpreter {
+  header: true;
+  columns: [
+    "mpg" oftype decimal,
+    "cyl" oftype integer,
+    "disp" oftype decimal,
+  ];
+}
+block CarsLoader oftype SQLiteLoader {
+  table: "Cars";
+  file: "./cars.sqlite";
+}
+```
+
+Composite blocks are a special case that use a smiliar pipe syntax to chain blocks. For the scope of this RFC, chaining pipeline syntax should **stay possible in pipelines and composite blocks** while **composite blocks are also able to contain block definitions**. In a future enhancement (see Possible Future Changes/Enhancements), this should be split up with the introduction of packages (ref [RFC0015](../0015-multi-file-jayvee/0015-multi-file-jayvee.md)).
+
+#### 4. Allow only one DAG per pipeline
+- If a user creates two graphs in one pipeline, the pipeline definition becomes invalid
+  - A Jayvee model can contain more than one pipeline definition
+  - All pipelines in a Jayvee model are instantiated and executed when the model is interpreted
+```
+// Valid pipeline
+pipeline examplePipeline {
+  A -> B -> C;
+  B -> D;
+}
+
+// Invalid pipeline
+pipeline examplePipeline {
+  A -> B -> C;
+  D -> E;
+}
+
+// Alternative to the invalid pipeline, create two pipelines
+pipeline examplePipeline1 {
+  A -> B -> C;
+}
+pipeline examplePipeline2 {
+  D -> E;
+}
+```
+
+## Drawbacks
+- With additional enforced structure, users loose flexibility to define things where they want
+
+## Alternatives
+- Introduce explicit execution calls like `pipeline.execute();` or `instantiate <pipeline name>;`/`run <pipeline name>;` instead of implicitly executing pipelines
+  - Decided against because Jayvee is purely descriptive right now outside of expressions
```

Comment:
We could go with calling it `declarative`. Or am I misinterpreting something here? https://en.wikipedia.org/wiki/Comparison_of_programming_paradigms
```suggestion
  - Decided against because Jayvee is purely declarative right now outside of expressions
```